crates/e2e/src/setup/solver/mod.rs

```diff
@@ -0,0 +1,16 @@
+use tokio::signal::{unix, unix::SignalKind};
+
+pub mod mock;
+pub mod solution;
+
+#[cfg(unix)]
+async fn shutdown_signal() {
+    // Intercept main signals for graceful shutdown.
+    // Kubernetes sends sigterm, whereas locally sigint (ctrl-c) is most common.
+    let mut interrupt = unix::signal(SignalKind::interrupt()).unwrap();
+    let mut terminate = unix::signal(SignalKind::terminate()).unwrap();
+    tokio::select! {
+        _ = interrupt.recv() => (),
+        _ = terminate.recv() => (),
+    };
+}
```

Comment:
Do we really need it? Since the server is started inside the test environment, all threads, including the background ones, are dropped, and neither sigint nor sigkill are sent. Just to make sure, try adding a log inside the tokio::select. I couldn't find any log there.

Reply:
Yes, as @MartinquaXD says, if you are running a local test in debug mode and you kill the process because you stop it/overflow/whatever, then the demon keeps running making the port unavailable, and then you need to find the process, kill it, etc.